### ISSUE TITLE
[Regression Tracker] PR #5785 (rsx: Texture cache fixes)

### ISSUE THREAD

**Issue by nitrohigito** [RPCS3/rpcs3#5869]
This is a ticket created for tracking regressions caused by #5785 (rsx: Texture cache fixes).

Noted regressions:

- [x] bottom third of the FBOs being black, see for example in the game loader of God of War: Collection [BCES00791] (issue #5864), or in the menus of Mobile Suit Gundam: Target in Sight [BLES00034] (issue #5847), Page Chronica (Demo) [NPHB00533], or Cash Guns Chaos [NPUB30001]

- [ ] partially black FBOs while ingame in NASCAR Unleashed [BLUS30806] and in F1 Race Stars [NPEB01124]; thought to be a memory barrier issue

- [x] ~~completely broken / missing graphics and crashes in various games caused by gpu blit issues, see issue #5807, #5826 and PR #5813~~ **[fixed by PR #5813]**

- [x] broken graphics with GL and crash with VK on opening the book (Wizards's Companion) in the main menu at Ni no Kuni: Wrath of the White Witch [BLUS30947] and in the menus of Motorcycle Club Demo [NPUB91024], see issue #5848 and here

- [ ] flickering crowd in Backbreaker Vengeance [NPEB00592], see issue #5842 (also tracked separately, as it's not just a regression, but also a progression)

To avoid clutter, you may try and do the following to make images hidden by default (good for image comparisons for example):
```
<details>
<summary>[press to expand]</summary>

regular markdown formatted content goes here (don't remove the blank line before this)
</details>
```

**Comment by nitrohigito** [comment]
**God of War: Collection [BCES00791] regression:**

The game selection menu now has about ~1/3rd of its output missing, similarly to #5847. This issue is also caused by #5785.

**Expected** (before #5785):

![image](https://user-images.githubusercontent.com/10439488/56432899-59736d80-62cf-11e9-8d77-891409c1ae17.png)

**Current** (after #5785):

![image](https://user-images.githubusercontent.com/10439488/56433029-d999d300-62cf-11e9-9d32-f438e2c49770.png)

Logs from before #5785: 
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/3098668/RPCS3.log.gz)

Logs from after #5785:
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/3098673/RPCS3.log.gz)

Couldn't take RRCs due to the fact that they were broken at the time, but I could take one with current master (v0.0.6-7990-777a99d0). Hopefully it'll suffice.
[BCES00791_20190419182307_capture.zip](https://github.com/RPCS3/rpcs3/files/3098679/BCES00791_20190419182307_capture.zip)

System information:
 - CPU: i7 7700HQ
 - RAM: 16 GB
 - GPU: GTX 1060M (6GB ver, driver version: 425.31 [latest])
 - OS: Windows 10 Home (version: 1809, build: 17763.437 [latest])

**Comment by legend800** [comment]
Half regression/half improvement from that PR :)
https://github.com/RPCS3/rpcs3/issues/5842

**Comment by uaqlover** [comment]
Regression Mobile Suit Gundam Target in Sight [BLES00034] https://github.com/RPCS3/rpcs3/pull/5785

0.0.6-7947

![FPS_ 36 09 _ Vulkan _ 0 0 6-7947 _ Mobile Suit Gundam Target in Sight  BLES00034  15_04_2019 10_09_02 PM](https://user-images.githubusercontent.com/45731484/56183389-41e46e00-5fcb-11e9-9e18-43fdccd30bc5.png)


0.0.6-7935

![FPS_ 37 95 _ Vulkan _ 0 0 6-7935 _ Mobile Suit Gundam Target in Sight  BLES00034  15_04_2019 10_12_22 PM](https://user-images.githubusercontent.com/45731484/56183505-b0293080-5fcb-11e9-9ac5-96e8a44c3375.png)

**Comment by kd-11** [comment]
The cause for this glitch is easy to work around but needs more research first to make sure we do not introduce hidden bugs. I'll get on it immediately after the current shaders PR is merged.

**Comment by k10kan** [comment]
Last good build #5797 (0.0.6-7935)
First bad build #5785 (0.0.6-7947)

Vulkan: VK_ERROR_DEVICE_LOST
OpenGL: gfx corruption

After PR #5785, when accessing the book from main menu (Wizards's Companion):

![the_book](https://user-images.githubusercontent.com/37659629/56459547-6ce90c00-6395-11e9-827a-6923dd6fec36.png)

  * Vulkan - After opening the book game freezes with the error RSX: class std::runtime_error
    thrown: Assertion Failed! Vulkan API call failed with unrecoverable error: Device
    lost (Driver crashed with unspecified error or stopped responding and
    recovered) (VK_ERROR_DEVICE_LOST). RPCS3 is not frozen and clean exit is possible.

  * OpenGL - Gfx gets distorted (sample image included). After closing the book,
    gfx returns to normal.

Same issue with all PPU/SPU Decoders. Logs done using PPU LLVM/SPU LLMV.

Vulkan:
![rpcs3-v0 0 6-7947-12dc3c18_win64-Vulkan](https://user-images.githubusercontent.com/37659629/56197005-38ddb600-6038-11e9-871c-f4346918f844.png)
[rpcs3-v0.0.6-7947-12dc3c18_win64-Vulkan.log.gz](https://github.com/RPCS3/rpcs3/files/3084004/rpcs3-v0.0.6-7947-12dc3c18_win64-Vulkan.log.gz)

OpenGL:
![rpcs3-v0 0 6-7947-12dc3c18_win64-OpenGL](https://user-images.githubusercontent.com/37659629/56197035-45620e80-6038-11e9-9afd-42a97ccf0dfa.png)
[rpcs3-v0.0.6-7947-12dc3c18_win64-OpenGL.log.gz](https://github.com/RPCS3/rpcs3/files/3084005/rpcs3-v0.0.6-7947-12dc3c18_win64-OpenGL.log.gz)

Also, same issue presists in latest PR #5792 (0.0.6-800).

**Comment by Thanatosus** [comment]
Hello !

I have some more information for Ni no Kuni issue, hope it can helps a bit...

With OpenGL NVIDIA 1060 6G (425.31.0.0) on
RPCS3 v0.0.6-8000-ae5a4b69 Alpha | HEAD | Firmware version: 4.84
Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz | 8 Threads | 15.94 GiB RAM | AVX+ | TSX

when opening the Wizarding Book you can have or the glitches with skyrocketed FPS (like explained by k10kan) but also an error with just a game freeze
[non_fatal_RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/3100439/non_fatal_RPCS3.log.gz)

or a fatal error with a total freeze of RCPS3 .
[fatal_RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/3100437/fatal_RPCS3.log.gz)

Hope my log files could help you with those bugs.

Best regards and big thank you for your work on RPCS3 !

**Comment by uaqlover** [comment]
also Page Chronica (Demo) [NPHB00533]

![FPS_ 252 73 _ Vulkan _ 0 0 6-8004 _ Page Chronica (Demo)  NPHB00533  22_04_2019 2_36_04 AM](https://user-images.githubusercontent.com/45731484/56494556-7e223d80-64a7-11e9-99b2-6c5a5c1d042c.png)

**Comment by uaqlover** [comment]
also Cash Guns Chaos [NPUB30001]

![FPS_ 194 51 _ Vulkan _ 0 0 6-8004 _ Cash Guns Chaos  NPUB30001  22_04_2019 11_23_35 PM](https://user-images.githubusercontent.com/45731484/56559193-b97b4580-6555-11e9-9b36-fc3ee4f51bc5.png)

**Comment by nihanthkandimalla** [comment]
Ni No Kuni - Wizard's companion page works perfectly with minor gfx glitches in version 0.0.6-7935.

**Comment by nitrohigito** [comment]
Still broken.

![image](https://user-images.githubusercontent.com/10439488/56746431-79e27280-677c-11e9-8389-7ceaf1e8155a.png)

**Comment by nihanthkandimalla** [comment]
Doesn't fix the Ni no Kuni - Book Issue.. on top of that log size got even bigger than before.
[RPCS3.zip](https://github.com/RPCS3/rpcs3/files/3117977/RPCS3.zip)

**Comment by k10kan** [comment]
No change in Ni no Kuni with **rpcs3-v0.0.0.6-8035-cc4a4914_win64** (rsx: Ignore transfer offsets when wrapping behaviour is expected):
* Vulkan: VK_ERROR_DEVICE_LOST
* OpenGL: gfx corruption

Log:
[RPCS3.zip](https://github.com/RPCS3/rpcs3/files/3118033/RPCS3.zip)


RRC (only OpenGL; Vulkan crashes):
[BLUS30947_20190425185742_capture_OpenGL.zip](https://github.com/RPCS3/rpcs3/files/3118028/BLUS30947_20190425185742_capture_OpenGL.zip)

**Comment by uaqlover** [comment]
also NASCAR® Unleashed [BLUS30806]

![FPS_ 28 94 _ OpenGL _ 0 0 6-8034 _ NASCAR® Unleashed  BLUS30806  27_04_2019 2_46_17 AM](https://user-images.githubusercontent.com/45731484/56847900-c0c48b00-6896-11e9-857f-b4a0e2d34919.png)

**Comment by uaqlover** [comment]
also Motorcycle Club Demo [NPUB91024] same as Ni no Kuni

![FPS_ 22 12 _ Vulkan _ 0 0 6-7935 _ Motorcycle Club Demo  NPUB91024  27_04_2019 5_03_15 AM](https://user-images.githubusercontent.com/45731484/56849531-751bdc80-68aa-11e9-8c05-11452ea5959f.png)

**Comment by uaqlover** [comment]
also Motorcycle Club Demo [NPUB91024] same as Ni no Kuni

![FPS_ 25 24 _ Vulkan _ 0 0 6-8034 _ Motorcycle Club Demo  NPUB91024  27_04_2019 5_12_48 AM](https://user-images.githubusercontent.com/45731484/56849587-3cc8ce00-68ab-11e9-8c12-8bfd62996ec6.png)

**Comment by kd-11** [comment]
Retest with https://ci.appveyor.com/project/kd-11/rpcs3/builds/24143201

**Comment by nitrohigito** [comment]
Fixes **God of War: Collection [BCES00791]**.

![image](https://user-images.githubusercontent.com/10439488/56850977-f191d800-6909-11e9-86ef-a01086c6e59f.png)

**Comment by uaqlover** [comment]
NASCAR® Unleashed [BLUS30806] still same 

![FPS_ 26 57 _ OpenGL _ 0 0 6-8036 _ NASCAR® Unleashed  BLUS30806  27_04_2019 7_37_22 AM](https://user-images.githubusercontent.com/45731484/56851071-b965a780-68bf-11e9-8467-2ad72fbfd5c5.png)

fixes Page Chronica (Demo) [NPHB00533]

![FPS_ 1465 19 _ Vulkan _ 0 0 6-8036 _ Page Chronica (Demo)  NPHB00533  27_04_2019 7_38_43 AM](https://user-images.githubusercontent.com/45731484/56851076-c8e4f080-68bf-11e9-9bfe-3ddce663253c.png)

also fixes Cash Guns Chaos [NPUB30001]

![FPS_ 752 97 _ Vulkan _ 0 0 6-8036 _ Cash Guns Chaos  NPUB30001  27_04_2019 7_39_18 AM](https://user-images.githubusercontent.com/45731484/56851077-d7330c80-68bf-11e9-9fe0-ea1b7af62e9f.png)

also fixes Mobile Suit Gundam Target in Sight [BLES00034] 

![FPS_ 39 29 _ Vulkan _ 0 0 6-8036 _ Mobile Suit Gundam Target in Sight  BLES00034  27_04_2019 7_39_54 AM](https://user-images.githubusercontent.com/45731484/56851081-e87c1900-68bf-11e9-8054-27bf3e127d67.png)

**Comment by nitrohigito** [comment]
I guess the NASCAR Unleashed issue really is separate then, updates the regressions list to say so.

What about Motorcycle Club Demo [NPUB91024]? Is it always like that or just in specific places? @uaqlover

**Comment by uaqlover** [comment]
@nitrohigito about Motorcycle Club i think only in menus ingame is fine still some graphical issue 
![FPS_ 22 59 _ Vulkan _ 0 0 6-8036 _ Motorcycle Club Demo  NPUB91024  27_04_2019 7_43_15 AM](https://user-images.githubusercontent.com/45731484/56851249-9a681500-68c1-11e9-9a27-c29e2a78bddf.png)


![FPS_ 28 95 _ Vulkan _ 0 0 6-8036 _ Motorcycle Club Demo  NPUB91024  27_04_2019 7_52_21 AM](https://user-images.githubusercontent.com/45731484/56851246-90461680-68c1-11e9-9e80-7272610e8b0c.png)

**Comment by kd-11** [comment]
Pretty sure ingame issue is unrelated to the regression.
Nascar unleashed bug is an obvious barrier problem, post openGL screenshot for comparison.

**Comment by nitrohigito** [comment]
Just tested it, works fine on AMD.

<details>
<summary>[press to expand]</summary>

![image](https://user-images.githubusercontent.com/10439488/56851917-dc6e7680-6914-11e9-8126-799a2c4b8190.png)
</details>

**Comment by kd-11** [comment]
Looks to be confirmed as a barrier problem, need a renderdoc trace to investigate.

**Comment by nitrohigito** [comment]
Just tested it on nvidia, the game behaves very strangely. 

On first ingame with AMD the screen was black, only the HUD was visible, but with maximizing/minimizing the window the actual ingame stuff appeared and i could play.

<details>
<summary>Image of the issue[press to expand]</summary>

![image](https://user-images.githubusercontent.com/10439488/56852394-5b66ad80-691b-11e9-9a7e-e027db49665f.png)
</details>

With nvidia however, the blackness persists, and then RPCS3 crashes along with a VK_DEVICE_LOST. Relaunching the game doesn't help it'll always remain this way, so I can't take RRCs of that issue @uaqlover  is describing (though, i guess it could be essentially the same thing).

Another curious thing is that the game runs super fast for me by default, but when framelimited to 60, it slows down to a crawling 15 fps. Super strange.

RRC: <https://mega.nz/#!A4VRgKZI!JiKuRNmJxLop0h3GTHTF1sRSX7QVA8g8jZ6BGhSU8ts>
Log: [RPCS3.log](https://github.com/RPCS3/rpcs3/files/3124098/RPCS3.log)

**Comment by uaqlover** [comment]
@nitrohigito it was always there the problem VK_ERROR_DEVICE_LOST with Vulkan on nvidia card https://github.com/RPCS3/rpcs3/issues/5659 i have 980 ti so it only works fine with opengl 

and the game runs super fast in menus only it will be locked in 30fps ingame 

and it was fine in RPCS3 v0.0.6-7935 with Vulkan 

<details>
<summary>[press to expand]</summary>

![FPS_ 29 97 _ Vulkan _ 0 0 6-7935 _ NASCAR® Unleashed  BLUS30806  27_04_2019 9_56_34 AM](https://user-images.githubusercontent.com/45731484/56852673-fc7d4600-68d2-11e9-82a3-9dedf36b173c.png)

</details>

i confirm the issue with black screen with new pr rpcs3-v0.0.0.6-8036-989c02e4_win64

<details>
<summary>[press to expand]</summary>

![FPS_ 30 01 _ Vulkan _ 0 0 6-8036 _ NASCAR® Unleashed  BLUS30806  27_04_2019 9_48_04 AM](https://user-images.githubusercontent.com/45731484/56852559-a8be2d00-68d1-11e9-8f93-58889164c68d.png)
</details>

**Comment by k10kan** [comment]
Ni no Kuni **rpcs3-v0.0.0.6-8036-989c02e4_win64** (rsx: Fix broken texture cache search when flipping) still not fixed:

* Vulkan: VK_ERROR_DEVICE_LOST
* OpenGL: gfx corruption

**Comment by nitrohigito** [comment]
Some update: Tested with AMD again, and on OpenGL. Curiously enough, I could reproduce uaq's issue:

<details>
<summary>[press to expand]</summary>

![image](https://user-images.githubusercontent.com/10439488/56854178-e010f600-6932-11e9-97f9-9506f3c7b72a.png)
</details>

RRC: <https://mega.nz/#!EsdiTKzC!tHz6muyIDBWgVL_LEGp6oO_ts3AGkavQbxFxrjc8Ug4>

**Comment by uaqlover** [comment]
also F1 Race Stars™ [NPEB01124]

<details>
<summary>OpenGL [press to expand]</summary>

![FPS_ 38 47 _ OpenGL _ 0 0 6-8036 _ F1 Race Stars™  NPEB01124  27_04_2019 12_32_05 PM](https://user-images.githubusercontent.com/45731484/56854265-ab784c80-68e8-11e9-9a03-3883c83b1eaf.png)

</details>

<details>
<summary>Vulkan [press to expand]</summary>

![FPS_ 59 93 _ Vulkan _ 0 0 6-8036 _ F1 Race Stars™  NPEB01124  27_04_2019 12_33_07 PM](https://user-images.githubusercontent.com/45731484/56854283-c2b73a00-68e8-11e9-8ec1-71ce7947cd57.png)

</details>

@nitrohigito can you confirm this issue with AMD card or it is only for nvidia card

**Comment by nitrohigito** [comment]
Sure, soon.

**Comment by nitrohigito** [comment]
Can confirm, also happens with AMD on VK. (latest test commit)

<details>
<summary>[press to expand]</summary>

![image](https://user-images.githubusercontent.com/10439488/56855596-ce871880-6949-11e9-9947-7aa8b725219a.png)</details>

RRC: https://mega.nz/#!4pMyTYYQ!9uqeD0rZoM7axzFUt95Bg70FIZQd2D27n8y5cNTiZpQ
Logs: [RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/3124401/RPCS3.log.gz)

The rsx capture loads back accurately this time, however, using VK, it crashes on my 1 gig HD7770 with an out of memory error. (the rsx capture viewer)

**Comment by Johndeep** [comment]
Apologize for being late with the report, but from the version _rpcs3-v0.0.6-7935-f30af3cc_linux64_ to _rpcs3-v0.0.6-7947-12dc3c18_linux64_, which is this the PR in question, the graphic glitch, the black bar in the lower half of the entire screen, in the game Hyperdimension Neptunia [BLUS30701] can be observed. Oh, and the blur effects are gone again.

![Bildschirmfoto_2019-04-28_11-23-37](https://user-images.githubusercontent.com/306953/56862147-205e8b80-69a8-11e9-9777-424c2005e3fd.png)
With the current version rpcs3-v0.0.6-8034-be6d9af1_linux64, this graphic issue remains a problem.

**Comment by uaqlover** [comment]
@Johndeep should be fixed by this pr build https://ci.appveyor.com/project/kd-11/rpcs3/builds/24143201

**Comment by Johndeep** [comment]
@uaqlover, oh, okay, wouldn't know this, but okay, since I'm relying on the releases available from https://rpcs3.net/compatibility?b. I'll test it again, if it is released there, thank you.

**Comment by kd-11** [comment]
It seems I only hid the problem instead of properly fixing it with my builds, needs a lot more work it seems.

**Comment by kd-11** [comment]
Should be fixed by https://github.com/RPCS3/rpcs3/pull/5895. Please test.

**Comment by k10kan** [comment]
Ni no Kuni: Wrath of the White Witch unfortunately **not** fixed with #5895 (rpcs3-v0.0.6-af915793_win64).

**Comment by Thanatosus** [comment]
Hello,

As k10kan says, Ni no Kuni is not fixed with #5895 (on OpenGL).

F {RSX [0x035cc88]} RSX: class std::runtime_error thrown: Unhandled GL format 0x822B
(in file c:\projects\rpcs3\rpcs3\emu\rsx\gl\glhelpers.h:1651)

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/3127429/RPCS3.log.gz)

**Comment by kd-11** [comment]
Confirmed out-of-bounds read for NNK, it seems to be unrelated to the other bugs reported here.

**Comment by kd-11** [comment]
NNK bug is fixed, at least RRC replays correctly.

**Comment by k10kan** [comment]
**Ni no Kuni: Wrath of the White Witch**

**kd-11**, you are on the right track. But now, beside already known problems, there are many errors like:

```E {RSX [0x1001042c]} RSX: Area merge failed! addr=0xc43c0b80, w=128, h=256, gcm_format=0x8b[sz=0]```


**OpenGL** - better, but not quite there
Two square areas are still wrong, and RPCS3 chrases upon exit.
I tried to create couple of captures, but every time upon loading the RRC file gfx corruption is not visable. (still I have included one, maybe you can use it anyway)

![opengl](https://user-images.githubusercontent.com/37659629/56912432-87b33300-6aaf-11e9-958c-6d4be050fc7b.png)

[rpcs3-v0.0.6-62acd588_win64_OpenGL.log.zip](https://github.com/RPCS3/rpcs3/files/3128346/rpcs3-v0.0.6-62acd588_win64_OpenGL.log.zip)
[BLUS30947_20190429183701_capture_OpenGL.zip](https://github.com/RPCS3/rpcs3/files/3128347/BLUS30947_20190429183701_capture_OpenGL.zip)


**Vulkan** - still VK_ERROR_DEVICE_LOST, does not crash upon RPCS3 exit.

![vulkan](https://user-images.githubusercontent.com/37659629/56912491-a9141f00-6aaf-11e9-981b-6fa61a73b3db.png)

[rpcs3-v0.0.6-62acd588_win64_Vulkan.log.gz](https://github.com/RPCS3/rpcs3/files/3128350/rpcs3-v0.0.6-62acd588_win64_Vulkan.log.gz)

**Comment by kd-11** [comment]
`E {RSX [0x1001042c]} RSX: Area merge failed! addr=0xc43c0b80, w=128, h=256, gcm_format=0x8b[sz=0]` is not really an error, its a diagnostic message.

As for vulkan crashing, are you using non-default settings or something? Like, maybe using resolutions scaling?

**Comment by k10kan** [comment]
As per all logs that I provided, settings are **always** same:

**RPCS3 default, both PPU/SPU LLVM, Resolution Scaling of 150%.**

Same settings are used accross the board of tested versions, starting from first issue where I detected
the regression. The last good build was #5797 (0.0.6-7935). Hence => the regression.

Just to make sure that it is unrelated to Resolution Scaling, I just tested vulkan renderer with (a) **Resolution Scaling of 100%** and (b) **with Strict Rendering Mode ON**.

Same behaviour, **same crash**. I can provide those logs too if needed.

**Comment by k10kan** [comment]
Also, as soon as book is being selected from menu, audio stutters like crazy for ~5 seconds and afterwards Vulkan crashes. Same audio stutter happens with OpenGL, but without crash, just gfx corruption.

Here is a video of log console, and by sound cue you will recognize moment when book has been opened.

[2019-04-29 19-20-00.zip](https://github.com/RPCS3/rpcs3/files/3128466/2019-04-29.19-20-00.zip)

**Comment by k10kan** [comment]
### Clean test - Ni no Kuni: Wrath of the White Witch
_Avoids any misconfiguration, shader/ppu/spu problems._

**All default settings with renderer changed to Vulkan.**

- Unpack **rpcs3-v0.0.6-62acd588_win64.7z**
- Create games.yml
- Symlink already installed game to dev_hdd0\game\BLUS30947
- Start RPCS3
- (all folders/files created, including home\00000001\savedata)
- Symlink already saved game to dev_hdd0\home\00000001\savedata\BLUS30947_00001
- Set log to trace
- Set XInput as player_1 controller
- **Only change Renderer to Vulkan**
- Install Firmware 4.84
- Run game
- Wait for PPU modules to compile...
- Wait for shaders to compile...
- Continue with saved position
- Load save position
- Wait for shaders to compile...
- Open menu (&#x25B3; - triangle)
- Navigate to book (&#8593; - up, &#8592; - left) and open book (X- cross)
- **FATAL ERROR: VK_ERROR_DEVICE_LOST**
- Stop game
- Exit RPCS3

All done in one go so all log messages are recorded in this one log file:

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/3128855/RPCS3.log.gz)

**Comment by Johndeep** [comment]
> Should be fixed by #5895. Please test.

![Bildschirmfoto_2019-04-30_11-13-29](https://user-images.githubusercontent.com/306953/56951869-318ad200-6b39-11e9-8a22-d653eb50b1ae.png)
With the PR-5895 from kd-11 on top of rpcs3-v0.0.6-8038-301dd828 (on Linux 64bit), the black bar is, at least in Hyperdimension Neptunia [BLUS30701], gone. Nice!

**Comment by kd-11** [comment]
@uaqlover Motorcycle Club was always broken, I have checked with builds from february and it was broken before. Actually confirm the regressions before reporting them, looking into stuff like this just wastes time as there is no meaningful reference frame provided.

**Comment by kd-11** [comment]
Same thing with F1 Race Stars. FFS
![image](https://user-images.githubusercontent.com/15904127/56958206-52abec80-6b52-11e9-91f2-6a5fff791a14.png)

**Comment by k10kan** [comment]
I was able to capture RDC of remaining **Ni no Kuni** gfx BUG in **OpenGL** with and without Strict Rendering Mode using newly released **RenderDoc 1.4** with **kd-11's 0.0.6-62acd588** build.

Vulkan crash since #5785 (0.0.6-7947), so no RDC is possible since.

_Files were too big, so I had to provide them using free file hosting service instead of github (10MB limit)!_

**OpenGL** - Clean install, **all settings default except _150% Resolution Scale_**
![RenderDoc_1 4_rpcs3-v0 0 6-62acd588_win64_OpenGL](https://user-images.githubusercontent.com/37659629/56970012-8f81de80-6b66-11e9-96c6-1ada2427c786.png)
File is 116MB, so use provided link (link valid for 30 days, download speed ~350KBps)
https://ufile.io/txf087yd

**OpenGL** - Clean install, **all settings default except _Strict Rendering Mode ON_**
![RenderDoc_1 4_rpcs3-v0 0 6-62acd588_win64_OpenGL-StrictRenderingMode](https://user-images.githubusercontent.com/37659629/56970024-93adfc00-6b66-11e9-8995-16579d846965.png)
File is 98MB, so use provided link (link valid for 30 days, download speed ~350KBps)
https://ufile.io/hp17v5ef


### ADDITIONAL INFO:
I figured out that this gfx BUG exists both in OpenGL and Vulkan (with or without SRM) and was only unmasked after kd-11's build 0.0.6-62acd588 solved pink texture corruption with current save game that I have (gfx bug exists also in 0.0.6-7935-f30af3cc, where game does not crash in Vulkan). I can provide save game if anyone who can help has Ni no Kuni game (I think it is legal to share saved games data, but IANAL).

I hope RDCs could help in fixing this gfx BUG. However, I can open new issue for gfx BUG only (present in both OpenGL/Vulkan) as it probably is not a regression anymore, but rather an old unmasked BUG. On the other hand, this gfx BUG has many similarities with **this** PR.

If all this sound confusing, I will state that after 0.0.6-62acd588 build opening Book started to work ok sometimes, but sometimes it produces already explained gfx BUG. 

**_Vulkan crash is regression as stated before._**

**Comment by kd-11** [comment]
Does this only trigger at some point in the game? I cannot trigger this with a new game on a 1050Ti.
![image](https://user-images.githubusercontent.com/15904127/56986585-c6221e00-6b93-11e9-94d2-28c2e37cc804.png)
Screen taken as soon as the wizards book is acquired. You may provide your saves for debugging if a special condition is needed to trigger this.

**Comment by k10kan** [comment]
kd-11, before you provided us with build 0.0.6-62acd588, pink texture bug happened everytime one would open the Book no matter when or where in the game. After 0.0.6-62acd588, pink texture was fixed, but this new anomally (two square wide) popped in. As I tried game using builds after #5785 (0.0.6-7947), this new savegame (that I am providing) was never tested when regression was reported, because I triggered pink texture bug super early in a game. Only yesterday I retested new savegame with #5785 (0.0.6-7947) and became aware that this bug is  actually not introduced with #5785 (0.0.6-7947), rather it existed before. Naturally, it means that I had not encountered this new bug first time I tested #5785 (0.0.6-7947).

Here is my savegame that triggers new bug as described in prior posts to this issue and for which I provided RDCs:
[Savegame_BLUS30947_00001.zip](https://github.com/RPCS3/rpcs3/files/3132796/Savegame_BLUS30947_00001.zip)

Also, I noticed that sometimes after opening and closing Book for 3-4 time rendering autofixes itself, but that behavior is not consistent.

And to answer your question directly, this new bug I trigger only when opening Book after loading from savegame, but truth to be told, I haven't progressed in the game further after I hit Vulkan bug, so I do not know does it happen only in savegame or also some other time.

**Comment by kd-11** [comment]
Turns out I needed your save to trigger this - on a new game even with a wizard book it did not trigger. Could be related to the map or something. Either way, the bug is properly fixed now.

**Comment by k10kan** [comment]
Great work **kd-11**!

I can confirm that with latest #5895 build 0.0.6-38bc23c9 OpenGL bug is gone. It came as an unexpected and very pleasant surprise that with the same build you also fixed Vulkan crash! Amazing!

@nitrohigito Can you please split Ni no Kuni issues from Motorcycle Club, and flag Ni no Kuni issues (OpenGL gfx bug and Vulkan crash) as **SOLVED**. (Motorcycle Club issue is probably resolved too, but IDK that; hopefully @uaqlover can check that one)

**Comment by nitrohigito** [comment]
Will be closed when merge-- and done :D
